File: binding/src/charts/libqarealegendmarker.cpp
#include <QAreaLegendMarker>
#include <QAreaSeries>
#include <QLegend>
#include <QLegendMarker>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <qarealegendmarker.h>
#include "libqarealegendmarker.h"
#include "libqarealegendmarker.hxx"

QAreaLegendMarker* QAreaLegendMarker_new(QAreaSeries* series, QLegend* legend) {
	 return new VirtualQAreaLegendMarker(series, legend);
}

QAreaLegendMarker* QAreaLegendMarker_new2(QAreaSeries* series, QLegend* legend, QObject* parent) {
	 return new VirtualQAreaLegendMarker(series, legend, parent);
}

libqt_string QAreaLegendMarker_Tr(const char* s) {
	QString _ret = QAreaLegendMarker::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

int QAreaLegendMarker_Type(QAreaLegendMarker* self) {
	return self->type();
}

QAreaSeries* QAreaLegendMarker_Series(QAreaLegendMarker* self) {
	return self->series();
}

libqt_string QAreaLegendMarker_Tr2(const char* s, const char* c) {
	QString _ret = QAreaLegendMarker::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QAreaLegendMarker_Tr3(const char* s, const char* c, int n) {
	QString _ret = QAreaLegendMarker::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

// Base class handler implementation
int QAreaLegendMarker_QBaseType(QAreaLegendMarker* self) {
	auto* vqarealegendmarker = dynamic_cast<VirtualQAreaLegendMarker*>(self);
	if (vqarealegendmarker && vqarealegendmarker->isVirtualQAreaLegendMarker) {
vqarealegendmarker->setQAreaLegendMarker_Type_IsBase(true);
	return vqarealegendmarker->type();
}
}

// Auxiliary method to allow providing re-implementation
void QAreaLegendMarker_OnType(QAreaLegendMarker* self, intptr_t slot) {
	auto* vqarealegendmarker = dynamic_cast<VirtualQAreaLegendMarker*>(self);
	if (vqarealegendmarker && vqarealegendmarker->isVirtualQAreaLegendMarker) {
vqarealegendmarker->setQAreaLegendMarker_Type_Callback(reinterpret_cast<VirtualQAreaLegendMarker::QAreaLegendMarker_Type_Callback>(slot));
}
}

// Base class handler implementation
QAreaSeries* QAreaLegendMarker_QBaseSeries(QAreaLegendMarker* self) {
	auto* vqarealegendmarker = dynamic_cast<VirtualQAreaLegendMarker*>(self);
	if (vqarealegendmarker && vqarealegendmarker->isVirtualQAreaLegendMarker) {
vqarealegendmarker->setQAreaLegendMarker_Series_IsBase(true);
	return vqarealegendmarker->series();
}
}

// Auxiliary method to allow providing re-implementation
void QAreaLegendMarker_OnSeries(QAreaLegendMarker* self, intptr_t slot) {
	auto* vqarealegendmarker = dynamic_cast<VirtualQAreaLegendMarker*>(self);
	if (vqarealegendmarker && vqarealegendmarker->isVirtualQAreaLegendMarker) {
vqarealegendmarker->setQAreaLegendMarker_Series_Callback(reinterpret_cast<VirtualQAreaLegendMarker::QAreaLegendMarker_Series_Callback>(slot));
}
}

void QAreaLegendMarker_Delete(QAreaLegendMarker* self) {
    delete self;
}

File: binding/src/charts/libqarealegendmarker.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qarealegendmarker.html
pub const qarealegendmarker = struct {

    /// New constructs a new QAreaLegendMarker object.
    pub fn New(series: ?*anyopaque, legend: ?*anyopaque) QtC.QAreaLegendMarker {
        return qtc.QAreaLegendMarker_new(@ptrCast(series), @ptrCast(legend));
    }


    /// New2 constructs a new QAreaLegendMarker object.
    pub fn New2(series: ?*anyopaque, legend: ?*anyopaque, parent: ?*anyopaque) QtC.QAreaLegendMarker {
        return qtc.QAreaLegendMarker_new2(@ptrCast(series), @ptrCast(legend), @ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAreaLegendMarker_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qarealegendmarker.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Type(self: ?*anyopaque, ) i32 {
        return qtc.QAreaLegendMarker_Type(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnType(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) i32) void {
        qtc.QAreaLegendMarker_OnType(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseType(self: ?*anyopaque, ) i32 {
        return qtc.QAreaLegendMarker_QBaseType(@ptrCast(self));
    }

    pub fn Series(self: ?*anyopaque, ) QtC.QAreaSeries {
        return qtc.QAreaLegendMarker_Series(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnSeries(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) QtC.QAreaSeries) void {
        qtc.QAreaLegendMarker_OnSeries(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSeries(self: ?*anyopaque, ) QtC.QAreaSeries {
        return qtc.QAreaLegendMarker_QBaseSeries(@ptrCast(self));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAreaLegendMarker_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qarealegendmarker.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAreaLegendMarker_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qarealegendmarker.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QAreaLegendMarker_Delete(@ptrCast(self));
    }
};
